improver_tests/ensemble_copula_coupling/test_utilities.py

```diff
@@ -329,177 +310,98 @@ class Test_insert_lower_and_upper_endpoint_to_1d_array(IrisTest):
     """Test the insert_lower_and_upper_endpoint_to_1d_array."""
 
     def test_basic(self):
-        """
-        Basic test that the result is a numpy array with the expected contents.
-        """
+        """Test that the result is a numpy array with the expected contents."""
         expected = [0, 20, 50, 80, 100]
-        percentiles = [20, 50, 80]
+        percentiles = np.array([20, 50, 80])
         result = insert_lower_and_upper_endpoint_to_1d_array(percentiles, 0, 100)
         self.assertIsInstance(result, np.ndarray)
         self.assertArrayAlmostEqual(result, expected)
 
-    def test_another_example(self):
-        """
-        Another basic test that the result is a numpy array with the
-        expected contents.
-        """
-        expected = [-100, -40, 200, 1000, 10000]
-        percentiles = [-40, 200, 1000]
-        result = insert_lower_and_upper_endpoint_to_1d_array(percentiles, -100, 10000)
-        self.assertIsInstance(result, np.ndarray)
-        self.assertArrayAlmostEqual(result, expected)
-
     def test_2d_example(self):
-        """
-        Another basic test that the result is a numpy array with the
-        expected contents.
-        """
+        """Test 2D input results in expected error"""
         percentiles = np.array([[-40, 200, 1000], [-40, 200, 1000]])
-        msg = "all the input arrays must have same number of dimensions"
+        msg = "Expected 1D input"
         with self.assertRaisesRegex(ValueError, msg):
             insert_lower_and_upper_endpoint_to_1d_array(percentiles, -100, 10000)
 
 
-class Test_restore_non_probabilistic_dimensions(IrisTest):
+class Test_restore_non_percentile_dimensions(IrisTest):
 
-    """Test the restore_non_probabilistic_dimensions."""
+    """Test the restore_non_percentile_dimensions."""
 
     def setUp(self):
         """Set up temperature cube."""
-        self.current_temperature_forecast_cube = set_up_percentile_cube(
+        self.cube = set_up_percentile_cube(
             np.sort(ECC_TEMPERATURE_REALIZATIONS, axis=0),
             np.array([10, 50, 90], dtype=np.float32),
         )
-
-    def test_basic(self):
-        """
-        Basic test that the result is a numpy array with the expected contents.
-        """
-        cube = self.current_temperature_forecast_cube
-        plen = len(cube.coord("percentile").points)
-        reshaped_array = restore_non_probabilistic_dimensions(
-            cube.data, cube, "percentile", plen
-        )
-        self.assertIsInstance(reshaped_array, np.ndarray)
+        # function is designed to reshape an input data array with dimensions of
+        # "percentiles x points" - generate suitable input data
+        points_data = [self.cube.data[i].flatten() for i in range(3)]
+        self.input_data = np.array(points_data)
```

Comment:
Should this be within the `test_percentile_is_dimension_coordinate` test, as it is only used in that test? Note that this is actually the same as:
```
self.input_data = self.cube.data.flatten()
```

Reply:
No this isn't the same as flattening the cube data.  `self.cube.data.flatten()` produces a 1D array with shape `(27,)`.  `self.input_data` has shape `(3, 9)`, which is "number of percentiles" x "number of (spatial) points", as stated in the comment.

I've ensured this array is used throughout rather than manipulating data from the template cube.